aapcs64-morello/aapcs64-morello.rst

```diff
@@ -753,53 +782,30 @@ The definition of ``va_list`` has implications for the internal implementation i
-
-
-Register Save Areas
--------------------
-
-The prologue of a function which accepts a variable argument list and which invokes the ``va_start`` macro is expected to save the incoming argument registers to three register save areas within its own stack frame: one area to hold the 64-bit general registers xn-x7, a second area to hold the 128-bit FP/SIMD registers vn-v7 and a third area to hold the capability registers cn-c7. Only parameter registers beyond those which hold the named parameters need be saved, and if a function is known never to accept parameters in registers of that class, then that register save area may be omitted altogether. In the first two areas the registers are saved in ascending order. The memory format of FP/SIMD registers save area must be as if each register were saved using the integer str instruction for the entire (ie Q) register.
+APPENDIX Variable argument Lists in AAPCS64-cap
+===============================================
 
+Languages such as C and C++ permit routines that take a variable number of arguments (that is, the number of parameters is controlled by the caller rather than the callee). Furthermore, they may then pass some or even all of these parameters as a block to further subroutines to process the list. If a routine shares any of its optional arguments with other routines then a parameter control block needs to be created. The remainder of this appendix is informative.
 
-The third register save area is located immediately above the general-purpose register area. The start of this area must be 16-byte aligned. It will contain the values of the capability registers cn-c7, stored starting at the address ``__gr_top`` from the highest numbered register to the lowest number register. All registers saved in this area must have their corresponding x sub-register stored in the general-purpose register save area as well. The capability register save area can be omitted if no capability arguments are used.
 
 The va_list type
 ----------------
 
-The ``va_list`` type may refer to any parameter in a parameter list, which depending on its type and position in the argument list may be in one of three memory locations: the current function’s general register argument save area, its FP/SIMD register argument save area, or the calling function’s outgoing stack argument area.
+The ``va_list`` type may refer to any parameter in a parameter list. All Anonymous parameters are passed on the stack in AAPCS64-cap.
 
 .. code-block:: c
 
     typedef struct  va_list {
-        void * stack; // next stack param
-        void * gr_top; // end of GP arg reg save area
-        void * vr_top; // end of FP/SIMD arg reg save area
-        int gr_offs; // offset from  gr_top to next GP register arg
-        int vr_offs; // offset from  vr_top to next FP/SIMD register arg
+        void * stack; // next param
```

Comment:
Or just make va_list itself be a `void *`? That's what other architectures do that don't have complex varargs tracking (e.g. RISC-V, for better or worse).